app/models/display_ad.rb

```diff
@@ -17,6 +17,7 @@ class DisplayAd < ApplicationRecord
                              inclusion: { in: ALLOWED_PLACEMENT_AREAS }
   validates :body_markdown, presence: true
   before_save :process_markdown
+  after_create :generate_display_ad_name
```

Comment:
Just curious: I was expecting it to be `before_save`. Is there a reason to prefer an `after_create` here?

Reply:
I did an `after_create` because I'm using the `id` of the ad in the display ad name if a name was not specified.

Reply:
I could use the `before_save` if i generate a random string like "Display Ad #{random string}" as the display_ad name instead hence removing the requirement on the `id` but I thought a random string is less memorable than a number. Do you have any thoughts on it?

Reply:
Right, that makes sense, I wasn't thinking about the `id` not being available `before`. Would we want it to be `after_save`, to avoid a scenario where someone could edit the ad and remove the name? (Or, we could maybe we make this some kind of attribute default, then we _always_ have a name, even if it's been cleared out? But I would imagine that's likely to have more edge cases.)

Reply:
Thanks for this @jaw6  - I went with the `after_save`,but to be honest I'm still feeling uncertain about whether the name really _needs_ the display_ad `id`.  ✨

The other option would be to just generate a random name before saving the Display Ad if the admin does not provide a name. That way, we won't have to make that second update update after we save which may be more optimal. However, I suspect that in most cases the admin will provide a name and so I'm trying not to overthink this detail but if other folks reviewing the PR feel more strongly about which way is the better way, I'm open to changing it accordingly.

Reply:
That makes sense to me :+1:

Reply:
It seems like we're not anticipating the name needing to follow any particular structure, the this default could be changed later without being a problem. So +1 on not over-thinking this.